<launch>
  <arg name="control_output" />
  <arg name="control_input" />
  <!-- Limits to 0.0015 if PWM, 0 otherwise -->
  <arg name="control_is_pulseTime" default="true" />
  <arg name="simulated" default="false" />
  
  <arg name="has_low_limit" default="false"/>
  <arg name="low_limit_pin" default="-1"/>
  
  <arg name="has_high_limit" default="false"/>
  <arg name="high_limit_pin" default="-1"/>

  <arg name="num" default="1" />

  <group ns="Arm_Board_HW" unless="$(arg simulated)">
    <node name="Switch$(arg num)_Low" pkg="mavric" type="PiGPIO_Input.py" if="$(arg has_low_limit)">
      <remap from="State" to="LimitSwitch$(arg num)_Low" />
      <param name="pin" value="$(arg low_limit_pin)" />
    </node>
    <node name="Switch$(arg num)_High" pkg="mavric" type="PiGPIO_Input.py" if="$(arg has_high_limit)">
      <remap from="State" to="LimitSwitch$(arg num)_High" />
      <param name="pin" value="$(arg high_limit_pin)" />
    </node>
  </group>

  <node name="Linac$(arg num)_Limits" pkg="mavric" type="LimitSwitch.py">
    <remap from="signal_in" to="$(arg control_input)" />
    <remap from="signal_out" to="$(arg control_output)" />
    <remap from="switch_low" to="Arm_Board_HW/LimitSwitch$(arg num)_Low" />
    <param name="use_switch_low" value="$(arg has_low_limit)" />

    <remap from="switch_high" to="Arm_Board_HW/LimitSwitch$(arg num)_High" />
    <param name="use_switch_high" value="$(arg has_high_limit)" />
    
    <param name="low_limit" value="0.0015" if="$(arg control_is_pulseTime)" />
    <param name="low_limit" value="0"      unless="$(arg control_is_pulseTime)" />
    
    <param name="high_limit" value="0.0015" if="$(arg control_is_pulseTime)" />
    <param name="high_limit" value="0"      unless="$(arg control_is_pulseTime)" />
  </node>

  <node name="linac$(arg num)_status" pkg="mavric" type="Actuator_Status.py" >
    <remap from="actuator_command" to="$(arg control_output)" />
    <remap from="switch_low" to="Arm_Board_HW/LimitSwitch$(arg num)_Low" />
    <remap from="switch_high" to="Arm_Board_HW/LimitSwitch$(arg num)_High" />
    <param name="movement_speed" value="0.04144" />
  </node>
</launch>
